MIGRATION.md

```diff
@@ -0,0 +1,222 @@
+- **azkustodata:** For querying and managing Azure Data Explorer clusters.
+- **azkustoingest:** For ingesting data into Azure Data Explorer clusters.
+
+### Importing the New Packages
+
+Depending on your requirements, import one or both of these packages into your Go files.
+
+**For Data Operations:**
+```go
+import (
+    "github.com/Azure/azure-kusto-go/azkustodata"
+)
+```
+
+**For Ingestion Operations:**
+```go
+import (
+    "github.com/Azure/azure-kusto-go/azkustoingest"
+)
+```
+
+### Using the New Packages
+
+Update your code to use the new packages.
+
+For exmaple:
+
+Old SDK:
+```go
+kustoConnectionStringBuilder := kusto.NewConnectionStringBuilder(endpoint)
+```
+
+New SDK (For Data Operations):
+```go
+kustoConnectionStringBuilder := azkustodata.NewConnectionStringBuilder(endpoint)
+```
+
+Same for ingestion operations:
+
+Old SDK Ingestion Client Creation:
+```go
+in, err := ingest.New(kustoClient, "database", "table")
+```
+
+New SDK Ingestion Client Creation (Queued Example):
+```go
+in, err := azkustoingest.New(kustoConnectionString)
+```
+
+## 3. Building Queries
+
+The new SDK introduces a new way to build queries.:
+The old SDK used a `kusto.NewStmt` method to build queries:
+```go
+    query := kusto.NewStmt("systemNodes | project CollectionTime, NodeId")
+```
+
+For the new SDK, use the `azkustodata/kql` package to build queries, which has a type-safe query builder:
+```go
+    dt, _ := time.Parse(time.RFC3339Nano, "2020-03-04T14:05:01.3109965Z")
+    tableName := "system nodes"
+    value := 1
+    
+    query := kql.New("")
+        .AddTable(tableName)
+        .AddLiteral(" | where CollectionTime == ").AddDateTime(dt)
+        .AddLiteral(" and ")
+        .AddLiteral("NodeId == ").AddInt(value) // outputs ['system nodes'] | where CollectionTime == datetime(2020-03-04T14:05:01.3109965Z) and NodeId == int(1)
+```
+
+## 4. Querying Data
+
+The new SDK introduces a new way to query data. 
+
+The old SDK used the `Query` method to query data:
+```go
+import github.com/Azure/azure-kusto-go/kusto/data/table
+
+// Query our database table "systemNodes" for the CollectionTimes and the NodeIds.
+iter, err := client.Query(ctx, "database", query)
+if err != nil {
+	panic("add error handling")
+}
+defer iter.Stop()
+
+// .Do() will call the function for every row in the table.
+err = iter.DoOnRowOrError(
+    func(row *table.Row, e *kustoErrors.Error) error {
+        if e != nil {
+            return e
+        }
+        if row.Replace {
+            fmt.Println("---") // Replace flag indicates that the query result should be cleared and replaced with this row
+        }
+        fmt.Println(row) // As a convenience, printing a *table.Row will output csv
+        return nil
+	},
+)
+if err != nil {
+	panic("add error handling")
+}
+```
+
+The new SDK returns a dataset object, with the ability to handle different tables as they come:
+```go
+dataset, err := client.QueryIterative(ctx, "database", query)
```

Comment:
So we want iterative to be the default user behavior ?

Reply:
I'd show a non-iterative and an iterative example in that order

Reply:
Better?

Reply:
Yes